app/autosynccontroller.h

```diff
@@ -29,40 +30,45 @@ class AutosyncController : public QObject
     {
       Synced = 0,
       SyncInProgress,
-      PendingChanges,
-      SyncKeepsFailing
+      PendingChanges, // awaits sync
+      SyncKeepsFailing,
+      NotAMerginProject
     };
-    Q_ENUMS( SyncStatus )
+    Q_ENUM( SyncStatus )
 
-    explicit AutosyncController( QObject *parent = nullptr );
+    explicit AutosyncController( LocalProject *openedProject, QgsProject *openedQgsProject, QObject *parent = nullptr );
 
     virtual ~AutosyncController();
 
     SyncStatus syncStatus();
 
-    void setActiveProject( Project project );
-
-    void setActiveQgsProject( QgsProject *qgsProject );
-
   signals:
 
-    void syncProject( Project *project );
+    void foundProjectChanges( const QString &projectNamespace, const QString &projectName );
 
     void syncStatusChanged( SyncStatus status );
 
   public slots:
 
-    void handleSyncFinished();
+    void synchronizationProgressed( const QString &projectFullName, qreal progress );
+
+    void synchronizationFinished( const QString &projectDir, const QString &projectFullName, bool successfully, int version );
+
+    void handleLocalChange();
 
   private:
+    void setSyncStatus( SyncStatus status );
 
     SyncStatus mSyncStatus = Synced;
 
-    Project *mActiveProject = nullptr;
-
-    QgsProject *mActiveQgsProject = nullptr;
+    LocalProject *mActiveProject = nullptr; // not owned
+    QgsProject *mActiveQgsProject = nullptr; // not owned
+    MerginApi *mBackend = nullptr; // not owned
```

Comment:
I usually prefer to call instance m+name_of_class. better readable in the cpp file usually